### ISSUE TITLE
LR finder advances scheduler per microbatch under gradient accumulation

### ISSUE THREAD

**Issue by frgfm** [frgfm/Holocron#504]
## Bug description

`Trainer.find_lr` advances its learning-rate scheduler once per **microbatch**, while the optimizer only steps once per `gradient_acc` microbatches.

This makes the LR range test invalid whenever `gradient_acc > 1`:

- the scheduler advances before an optimizer step for the first `gradient_acc - 1` microbatches;
- the scheduler advances `gradient_acc` times faster than the optimizer;
- `lr_recorder` associates each microbatch loss with a nominal LR even though no optimizer update occurred at most of those points;
- a final partial accumulation is never flushed when `num_it % gradient_acc != 0`;
- under AMP, the scheduler can also advance when `GradScaler` skips the optimizer step.

The issue is visible on current `main` and remains in `codex/yolov4-training-repair`. Normal epoch training on the repair branch already ties scheduler advancement to successful optimizer steps, but `find_lr` still has its own unconditional `scheduler.step()`.

Relevant code:

- [`find_lr` calls `_backprop_step` and then unconditionally advances the scheduler](https://github.com/frgfm/holocron/blob/200c0944098b30cd6199c654c425ddfac0e22072/holocron/trainer/core.py#L378-L380)
- [`_backprop_step` only updates after `gradient_acc` microbatches](https://github.com/frgfm/holocron/blob/200c0944098b30cd6199c654c425ddfac0e22072/holocron/trainer/core.py#L203-L227)
- [`lr_recorder` is currently sized and labelled per microbatch](https://github.com/frgfm/holocron/blob/200c0944098b30cd6199c654c425ddfac0e22072/holocron/trainer/core.py#L366-L392)

For example, `num_it=100` and `gradient_acc=8` currently produce:

- 100 scheduler steps;
- 12 optimizer steps;
- four pending microbatches discarded when the method returns.

The resulting plot does not represent loss at the learning rates actually used by optimizer updates and can recommend a misleading maximum LR.

## Reproduction

Run the detection LR finder with gradient accumulation:

```bash
cd references/detection
python train.py VOC2012 \
  --arch yolov4 --img-size 608 \
  -b 8 --grad-acc 8 -j 8 \
  --opt sgd --freeze-until backbone \
  --amp --device 0 --find-lr
```

The same problem can be reproduced with a small toy `Trainer`, a counting optimizer, `gradient_acc=8`, and `num_it=100`: count calls to `optimizer.step()` and `scheduler.step()` and compare them with the recorded LR/loss points.

Depending on the PyTorch version, the run may also emit:

```text
Detected call of `lr_scheduler.step()` before `optimizer.step()`
```

No exception is required for the bug to occur; the primary failure is an incorrect LR-versus-loss curve.

## Expected behavior

Every recorded LR/loss point should correspond to an optimizer update, and the scheduler should advance only after a successful optimizer step.

Preserve the current meaning of `num_it` as the maximum number of consumed microbatches:

1. Size the LR sweep using `ceil(num_it / gradient_acc)` optimizer steps.
2. Keep the LR unchanged throughout each accumulation window.
3. Record the LR actually used for the optimizer update and one representative/averaged loss for that window.
4. Flush a final partial accumulation using its actual microbatch count.
5. Advance the scheduler only when the optimizer really stepped, including under AMP overflow handling.
6. Keep `gradient_acc=1` behavior backward compatible.

If `num_it` is instead redefined as optimizer steps, that semantic change must be documented and the loader-bound validation updated accordingly.

## Acceptance tests

- With `gradient_acc=1`, optimizer-step count, scheduler-step count, and recorder lengths remain aligned.
- With `gradient_acc=2` and an odd `num_it`, the final partial accumulation is applied.
- With `gradient_acc=8` and `num_it=100`, scheduler-step count equals optimizer-step count rather than microbatch count.
- Recorded LR values are the values actually used by the corresponding optimizer updates.
- No scheduler-before-optimizer warning is emitted.
- A skipped AMP optimizer step does not advance the scheduler or add a misleading recorder point.
- Existing `plot_recorder` behavior remains valid because `lr_recorder` and `loss_recorder` have equal, non-zero lengths.

## Environment

This is device- and dataset-independent control-flow behavior in `Trainer.find_lr`. It was observed while running YOLOv4 on Colab with Python 3.12, PyTorch CUDA, `gradient_acc=8`, and AMP, and confirmed by source inspection on `main` commit `200c0944098b30cd6199c654c425ddfac0e22072`.
